### ISSUE TITLE
Check files in the vip-config/ directory for WordPress functions

### ISSUE THREAD

**Issue by GaryJones** [Automattic/VIP-Coding-Standards#467]
## What problem would the enhancement address for VIP?

The VIP app repo has a [`vip-config/` directory](https://github.com/Automattic/vip-go-skeleton/tree/master/vip-config) which contains files that run before WordPress is loaded. Usually there's just a `vip-config/vip-config.php` file, but clients can create other files in this directory and include/require them from the default file.

Since WordPress is not loaded when these files are read, WordPress-defined functions are not available, but this can be easy to forget. Using a WP function when not defined, gives a fatal error which can take down a site.

## Describe the solution you'd like

A new sniff that can be applied to files in the `vip-config/` directory, and look for any functions that are not defined yet. We can use [`get_defined_functions()` function](https://www.php.net/manual/en/function.get-defined-functions.php) to look for defined functions. The `internal` index key will verify against PHP native functions, but we also need to consider functions that may be defined within the files of `vip-config/` itself. It may also depend on if a function call is done before or after the definition in the same file. There may be some false positives cases, but these are more likely to be edge cases vs capturing functions only defined in WordPress.

Early in the `process()`, use `$phpcs->getFilename()` which can give the complete file path, to abort the sniff early if not correct. The sniff should have a public property with the default pattern of files which should be scanned.

## What code should be reported as a violation?

```php
// In vip-config/vip-config.php:
home_url();
wp_redirect( ... );
// etc.
```

## What code should *not* be reported as a violation?

```php
// In vip-config/vip-config.php:
function foo() {
	return 'foo';
}
foo();

bar();
function bar() {
	return 'bar';
}
```

**Comment by GaryJones** [comment]
If someone creates a custom function in that file, and then calls it, it’s not going to be in the `get_defined_functions()` list, so we will need to have Warnings / false positives in this case.

**Comment by GaryJones** [comment]
`add_filter` and `add_action` would be another two good functions to look for.
